build.sh

```diff
@@ -91,12 +92,20 @@ if [ "$LINK_TYPE" == "shared" ]; then
 fi
 
 # Set target MacOS minver
-if [ "$MAC_ARCH" == "arm64" ]; then
+if [[ $3 == MACOSX_DEPLOYMENT_TARGET* ]] || [[ $4 == MACOSX_DEPLOYMENT_TARGET* ]] || [[ $5 == MACOSX_DEPLOYMENT_TARGET* ]]; then
```

Comment:
This part of code can be simplified by letting user optionally set the env-variable `MACOSX_DEPLOYMENT_TARGET` (just like CMAKE_OPTS and GIT_PULL_TOKEN) from shell/terminal before invoking this script. And then we can just write:

```bash
default_mac_os_target=$([ "$MAC_ARCH" == "arm64" ] && echo "11.10" || echo "10.10")
[ -z "$MACOSX_DEPLOYMENT_TARGET" ] && export MACOSX_DEPLOYMENT_TARGET="${default_mac_os_target}"
echo "macosx deployment target="$MACOSX_DEPLOYMENT_TARGET
```

The help comment above can be changed to mention `MACOSX_DEPLOYMENT_TARGET` under env-variable section